### ISSUE TITLE
Lazy inner tags are not shown

### ISSUE THREAD

**Issue by kethan** [yisar/fre#276]
`import { lazy, Suspense } from "fre"
const Lazy = lazy(() => import('./Hello'));
export function App() {
  return (
    <div>
      <Suspense fallback={'loading'}>
        <Lazy />
        <div>111</div>
        <Lazy />
        <div>111</div>
      </Suspense>
    </div>
  )
}`
        This `<div>111</div>` is not shown.

**Comment by yisar** [comment]
What version do you use? This looks like a bug, and I'll refactor them in v2.1.

**Comment by kethan** [comment]
> What version do you use? This looks like a bug, and I'll refactor them in v2.1.

  "dependencies": {
    "fre": "^2.0.7"
  },

**Comment by yisar** [comment]
Thank you. I'll refactor them in 2.1. I've come up with a better idea

Wait for me.

**Comment by kethan** [comment]
> Thank you. I'll refactor them in 2.1. I've come up with a better idea
> 
> Wait for me.

ok, thanks. I my office project thinking to use fre very soon. Is there a context provider in fre as in react?

**Comment by yisar** [comment]
> Is there a context provider in fre as in react?

https://github.com/yisar/fre/issues/213#issuecomment-754768434

I haven't implemented context selector yet, but it can be implemented externally through pub-sub simulation

**Comment by kethan** [comment]
> > Is there a context provider in fre as in react?
> 
> [#213 (comment)](https://github.com/yisar/fre/issues/213#issuecomment-754768434)
> 
> I haven't implemented context selector yet, but it can be implemented externally through pub-sub simulation

Thanks, I made context working.
```ts
import { useState } from "fre"
import { createContext, useContext } from "./context";
const Theme = createContext('light');

function NestedTheme() {
    const theme = useContext(Theme);
    return <p>Nested Active theme: {theme}</p>;
}

function DisplayTheme(props) {
    const theme = useContext(Theme);
    return (
        <div>
            {props && props.children}
            <p>Display Active theme: {theme}</p>
        </div>
    )
}

export default () => {
    const [count, setCount] = useState(0)
    return (
        <div>
            <h1>{count}</h1>
            <button onClick={() => setCount(count + 1)}>+</button>
            <Theme.Provider value="light">
                <DisplayTheme>
                    <NestedTheme />
                </DisplayTheme>
                <DisplayTheme />
            </Theme.Provider>
        </div>)
}
```
But getting issues while displaying children. 
fre.js:1 Uncaught TypeError: Cannot set property 'parent' of undefined

**Comment by kethan** [comment]
Any update on context or suspense? I have very less time i have to select one framework and start my office project.

**Comment by yisar** [comment]
I'm really sorry, because I'm reconstructing the core algorithm of fre, and a lot of things have changed.

I've released an emergency version that fixes context bugs, which you can do this:

```shell
yarn add fre@2.1.0-alpha
```
About lazy and Suspense, I haven't found a good way yet, maybe I will add them back in version 2.2.

You can temporarily replace them with useEffect

**Comment by kethan** [comment]
> th useEffect

How to do the same in useEffect()  can you share with me a code snippet, please?

**Comment by yisar** [comment]
```js
function App(){
  const [vdom, setVdom] = useState(null)
  useEffect(() =>{
     import('./app.js').then(({default})=>{
        setVdom(default)
     })
  })
  return <div>{vdom}</div>
}
```

**Comment by kethan** [comment]
> ```js
> function App(){
>   const [Component, setComponent] = useState(‘loading’)
>   useEffect(() =>{
>      import('./app.js').then(({default})=>{
>         setComponent(default)
>      })
>   })
>   return <div><Component/></div>
> }
> ```
> 
> It's not a good idea, but I'm still working on the implementation of Suspense, or have you noticed.

I tried context it's working fine. But the above code is loading the component but the button count is not updating.

**Comment by yisar** [comment]
@kethan Yes, it is different from Suspense. I will study the implementation of Suspense as soon as possible

**Comment by kethan** [comment]
> @kethan Yes, it is different from Suspense. I will study the implementation of Suspense as soon as possible

ok Thanks.

**Comment by kethan** [comment]
In preact, I am able to lazy load the component and then pass context to the lazy-loaded component.

```js
 <Suspense fallback={<div>loading...</div>}>
        <div>
        <Theme.Provider value="green">
          <Lazy />
        </Theme.Provider>
        </div>
  </Suspense>
```

**Comment by yisar** [comment]
I know this, preact external implementation, I also want to do this, will not put it in the core

But it's not easy to implement it safely in fre's current algorithm. I need time.

**Comment by kethan** [comment]
> I know this, preact external implementation, I also want to do this, will not put it in the core
> 
> But it's not easy to implement it safely in fre's current algorithm. I need time.

OK, sure no problem. meanwhile will start using preact now. Thanks for your support.

**Comment by kethan** [comment]
just curious any progress?

**Comment by yisar** [comment]
> just curious any progress?

I'm too busy at work.

**Comment by kethan** [comment]
> > just curious any progress?
> 
> I'm too busy at work.

Thanks for updating
